MureithiC/Twin-Towers
# As a supervisor I want to update non-sensitive information

**What does this PR do?**

1. Implements a feature that allows supervisors to update non-sensitive member information via a PUT /members/{id} API endpoint.
2. Restricts supervisors from modifying sensitive fields such as role and isActive status.
3. Adds validation to ensure only authorized supervisors can perform the updates.

**What are the tasks to be completed?**

1. Create a PUT /members/{id} endpoint for updating member information.
2. Implement validation to check the supervisor's role and ensure they cannot modify sensitive fields like role or isActive.
3. Allow only the following fields to be updated by supervisors:

- name
- phone
- email

4. Return a 200 OK status on successful update or an appropriate error response if the validation fails.
5. Write unit tests to verify the functionality, ensuring that sensitive fields cannot be modified.

**How can this be manually tested?**
Prerequisite: Supervisor must be authenticated.
Send a PUT request to /members/{id} with a JSON body containing updated member data, such as:
json

{
  "name": "Cyrus Mureithi",
  "phone": "+254729000000",
  "email": "abc@gmail.com"
}
The system should:
Return 200 OK if the update is successful.
Return an error if an attempt is made to update sensitive fields like role
Ensure that the updated fields (name, phone, email) are correctly stored in the database.

**Screenshots**
<img width="1440" alt="Screenshot 2024-10-09 at 13 05 42" src="https://github.com/user-attachments/assets/4d4283cc-aad4-4225-bd13-45c53e64a7d2">

<img width="1440" alt="Screenshot 2024-10-09 at 13 05 11" src="https://github.com/user-attachments/assets/dec064af-ac9c-4e36-8cca-0bd24abdcce4">
